Code/Framework/AzCore/Tests/Serialization/Json/TestCases_Patching.cpp

```diff
@@ -53,12 +56,27 @@ namespace JsonSerializationTests
             ASSERT_FALSE(patchDocument.HasParseError());
 
             AZ::JsonSerializationResult::ResultCode result = AZ::JsonSerialization::ApplyPatch(*m_jsonDocument,
-                m_jsonDocument->GetAllocator(), patchDocument, AZ::JsonMergeApproach::JsonPatch);
+                m_jsonDocument->GetAllocator(), patchDocument, AZ::JsonMergeApproach::JsonPatch, settings);
             EXPECT_EQ(result.GetTask(), AZ::JsonSerializationResult::Tasks::Merge);
             EXPECT_EQ(result.GetOutcome(), outcome);
             EXPECT_EQ(result.GetProcessing(), processing);
         }
 
+        void CheckApplyPatchOutcome(
+            const char* target,
+            const char* patch,
+            const char* expectedPatchedResult,
+            AZ::JsonSerializationResult::Outcomes outcome,
+            AZ::JsonSerializationResult::Processing processing,
+            const AZ::JsonApplyPatchSettings& settings = AZ::JsonApplyPatchSettings{})
+        {
+            CheckApplyPatchOutcome(target, patch, outcome, processing, settings);
+            rapidjson::Document expectedPatchedDocument;
+            expectedPatchedDocument.Parse(expectedPatchedResult);
+            ASSERT_FALSE(expectedPatchedDocument.HasParseError());
+            EXPECT_EQ(AZ::JsonSerialization::Compare(expectedPatchedDocument, *m_jsonDocument), AZ::JsonSerializerCompareResult::Equal);
```

Comment:
There's a utility function to compare json documents as strings. This will handle things like whitespaces, etc.